dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.cpp

```diff
@@ -281,91 +227,76 @@ bool ColumnFilePersistedSet::checkAndIncreaseFlushVersion(size_t task_flush_vers
-        for (auto & file : level)
-            new_level.push_back(file);
+        new_persisted_files.push_back(file);
+    }
+    for (const auto & file : column_files)
+    {
+        new_persisted_files.push_back(file);
     }
-    if (new_persisted_files_levels.empty())
-        new_persisted_files_levels.emplace_back();
-    auto & new_level_0 = new_persisted_files_levels[0];
-
-    for (const auto & f : column_files)
-        new_level_0.push_back(f);
-
     /// Save the new metadata of column files to disk.
-    serializeColumnFilePersistedLevels(wbs, metadata_id, new_persisted_files_levels);
+    serializeColumnFilePersisteds(wbs, metadata_id, new_persisted_files);
     wbs.writeMeta();
 
     /// Commit updates in memory.
-    persisted_files_levels.swap(new_persisted_files_levels);
+    persisted_files.swap(new_persisted_files);
     updateColumnFileStats();
-    LOG_DEBUG(log, "{}, after append {} column files, level info: {}", info(), column_files.size(), levelsInfo());
+    LOG_DEBUG(log, "{}, after append {} column files, persisted column files: {}", info(), column_files.size(), detailInfo());
 
     return true;
 }
 
 MinorCompactionPtr ColumnFilePersistedSet::pickUpMinorCompaction(DMContext & context)
 {
-    // Every time we try to compact all column files in a specific level.
+    // Every time we try to compact all column files.
     // For ColumnFileTiny, we will try to combine small `ColumnFileTiny`s to a bigger one.
-    // For ColumnFileDeleteRange and ColumnFileBig, we will simply move them to the next level.
-    // And only if there exists some small `ColumnFileTiny`s which can be combined together, we will actually do the compaction.
-    size_t check_level_num = 0;
-    while (check_level_num < persisted_files_levels.size())
+    // For ColumnFileDeleteRange and ColumnFileBig, we keep them intact.
+    // And only if there exists some small `ColumnFileTiny`s which can be combined, we will actually do the compaction.
+    auto compaction = std::make_shared<MinorCompaction>(minor_compaction_version);
+    if (!persisted_files.empty())
     {
-        check_level_num += 1;
-        if (next_compaction_level >= persisted_files_levels.size())
-            next_compaction_level = 0;
-
-        auto compaction = std::make_shared<MinorCompaction>(next_compaction_level, minor_compaction_version);
-        auto & level = persisted_files_levels[next_compaction_level];
-        next_compaction_level++;
-        if (!level.empty())
+        bool is_all_trivial_move = true;
+        MinorCompaction::Task cur_task;
+        for (auto & file : persisted_files)
         {
-            bool is_all_trivial_move = true;
-            MinorCompaction::Task cur_task;
-            for (auto & file : level)
+            auto pack_up_cur_task = [&]() {
+                bool is_trivial_move = compaction->packUpTask(std::move(cur_task));
+                is_all_trivial_move = is_all_trivial_move && is_trivial_move;
```

Comment:
Is it expected that those two lines duplicate with 295 and 296?

Reply:
Yes, it can be simplified.